pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -250,43 +235,60 @@ func (s *ServiceAntiAffinity) CalculateAntiAffinityPriorityMap(pod *v1.Pod, meta
 // The label to be considered is provided to the struct (ServiceAntiAffinity).
 func (s *ServiceAntiAffinity) CalculateAntiAffinityPriorityReduce(pod *v1.Pod, meta interface{}, sharedLister schedulerlisters.SharedLister, result framework.NodeScoreList) error {
+	reduceResult := make([]float64, len(result))
+	for _, label := range s.labels {
+		if err := s.updateNodeScoresForLabel(sharedLister, result, reduceResult, label); err != nil {
+			return err
+		}
+	}
+
+	// Update the result after all labels have been evaluated.
+	for i, nodeScore := range reduceResult {
+		result[i].Score = int64(nodeScore)
+	}
+	return nil
+}
+
+// updateNodeScoresForLabel updates the node scores for a single label. Note it does not update the
+// original result from the map phase directly, but instead updates the reduceResult, which is used
+// to update the original result finally. This makes sure that each call to updateNodeScoresForLabel
+// receives the same mapResult to work with.
+func (s *ServiceAntiAffinity) updateNodeScoresForLabel(sharedLister schedulerlisters.SharedLister, mapResult framework.NodeScoreList, reduceResult []float64, label string) error {
 	var numServicePods int64
-	var label string
+	var labelValue string
 	podCounts := map[string]int64{}
 	labelNodesStatus := map[string]string{}
 	maxPriorityFloat64 := float64(framework.MaxNodeScore)
 
-	for _, hostPriority := range result {
+	for _, hostPriority := range mapResult {
 		numServicePods += hostPriority.Score
 		nodeInfo, err := sharedLister.NodeInfos().Get(hostPriority.Name)
 		if err != nil {
 			return err
 		}
-		if !labels.Set(nodeInfo.Node().Labels).Has(s.label) {
+		if !labels.Set(nodeInfo.Node().Labels).Has(label) {
 			continue
 		}
 
-		label = labels.Set(nodeInfo.Node().Labels).Get(s.label)
-		labelNodesStatus[hostPriority.Name] = label
-		podCounts[label] += hostPriority.Score
+		labelValue = labels.Set(nodeInfo.Node().Labels).Get(label)
+		labelNodesStatus[hostPriority.Name] = labelValue
+		podCounts[labelValue] += hostPriority.Score
 	}
 
 	//score int - scale of 0-maxPriority
 	// 0 being the lowest priority and maxPriority being the highest
-	for i, hostPriority := range result {
-		label, ok := labelNodesStatus[hostPriority.Name]
+	for i, hostPriority := range mapResult {
+		labelValue, ok := labelNodesStatus[hostPriority.Name]
 		if !ok {
-			result[i].Name = hostPriority.Name
-			result[i].Score = 0
 			continue
 		}
 		// initializing to the default/max node score of maxPriority
 		fScore := maxPriorityFloat64
 		if numServicePods > 0 {
-			fScore = maxPriorityFloat64 * (float64(numServicePods-podCounts[label]) / float64(numServicePods))
+			fScore = maxPriorityFloat64 * (float64(numServicePods-podCounts[labelValue]) / float64(numServicePods))
 		}
-		result[i].Name = hostPriority.Name
-		result[i].Score = int64(fScore)
+		// The score of current label only accounts for 1/len(s.labels) of the total score.
```

Comment:
Say also that we compensate that by artificially multiplying the weight by the number of labels.